### PR TITLE
Added ability to get full list of labels

### PR DESCRIPTION
**Pull request by svermeulen**
For issue #913

### CODE REVIEW

File: jira/client.py [hunk 1]
```diff
@@ -3544,6 +3544,17 @@ def projectcategories(self):
 
         return data
 
+    @lru_cache(maxsize=None)
+    def labels(self):
+
+        url = self._options["server"] + "/rest/api/3/label"
```
Comment (adehad) [inline jira/client.py:3550]: Would transition this to use `self._get_json()`

**Review by ssbarnea — CHANGES_REQUESTED** [review]
Using print does not seem like ready for review code.

**Comment by studioj** [comment]
@svermeulen merge in master and resolve conflicts please

**Comment by studioj** [comment]
@svermeulen any idea if this will be picked up again? => our CI system is working again properly

**Comment by svermeulen** [comment]
Ok I rebased my branch to latest main, and removed the print statement.  Though I don't know how to address @adehad's comment

**Review by adehad — CHANGES_REQUESTED** [review]
This endpoint is only available on v3 of the API for Jira Cloud, this needs to be handled correctly.

`self._options["rest_api_version"]` has a value that can be checked against. 

Or we can instead check the `self._is_cloud` value and if true, use a URL with the v3 url?
That would require extension of the `_get_json()` and `_get_url()` methods to accept a version override.


```python
if not self._is_cloud:
    raise NotImplementedError("Only available on Jira Cloud instances.")
if str(self._options["rest_api_version"]) != "3":
    raise NotImplementedError("Needs v3 of the REST API Version.")

```

File: jira/client.py (hunk 1 above)
Comment (adehad) [reply jira/client.py:3968]: added an example in https://github.com/pycontribs/jira/pull/917#discussion_r739844741

**Comment by studioj** [comment]
no response after +1yr after requesting changes => close

**Comment by adehad** [comment]
@dvolosnykh I suggested some changes that were not addressed at the time, if you are able to add them in a new PR that would be ideal.

A test would be greatly appreciated too.

**Pull request closed without merging**